src/ngraph/op/slice_output.cpp

```diff
@@ -0,0 +1,137 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/slice_output.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+const string op::SliceOutput::type_name("SliceOutput");
+
+op::SliceOutput::SliceOutput(const Output<Node>& value,
+                             std::ptrdiff_t start,
+                             std::ptrdiff_t stride,
+                             std::ptrdiff_t part_size,
+                             std::ptrdiff_t end,
+                             std::ptrdiff_t axis)
+    : Op({value})
+    , m_start(start)
+    , m_stride(stride)
+    , m_part_size(part_size)
+    , m_end(end)
+    , m_axis(axis)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::SliceOutput::validate_and_infer_types()
+{
+    // For now, keep things simple; no padding, forwards stepping, etc.
+    // Later, use negative values for positioning from the end and working backwards.
+    NODE_VALIDATION_CHECK(this, 0 <= m_axis, "Negative axis is not supported.");
+    NODE_VALIDATION_CHECK(this, 0 <= m_start, "Negative start is not supported.");
+    NODE_VALIDATION_CHECK(this, 0 <= m_end, "Negative end is not supported.");
+    NODE_VALIDATION_CHECK(this, 1 <= m_part_size, "Part size must be positive.");
+    NODE_VALIDATION_CHECK(this, 1 <= m_stride, "Stride must be positive.");
+
+    PartialShape result_shape;
+    const PartialShape& value_shape = get_input_partial_shape(0);
+
+    if (value_shape.is_dynamic())
+    {
+        result_shape = PartialShape::dynamic();
+    }
+    else
+    {
+        vector<Dimension> value_dimensions = static_cast<vector<Dimension>>(value_shape);
+        Dimension sequence_dimension = value_dimensions[0];
+        value_dimensions.erase(value_dimensions.begin());
+
+        NODE_VALIDATION_CHECK(this,
+                              m_axis < value_dimensions.size(),
+                              "Axis is not within the sequence element shape.");
+        Dimension value_dimension = value_dimensions.at(m_axis);
+
+        if (sequence_dimension.is_dynamic() || value_dimension.is_dynamic())
```

Comment:
Please, answer the same questions like for `sliced_input`.

Reply:
Sliced input goes away, hopefully tomorrow. It is folded into `set_sliced_input`.